include/AST/NodeKind.def

```diff
@@ -0,0 +1,49 @@
+#ifndef NODE_KIND
+    #error "NODE_KIND must be defined"
+    #define NODE_KIND(Name)
```

Comment:
will need parent here too, and everywhere (can't implement the visitor otherwise)